generator/svelte/src/lib/service.ts

```diff
@@ -0,0 +1,27 @@
+import { PUBLIC_SUBSCRIPTIONS_BASE_URL } from '$env/static/public';
+import { toast } from './stores/toast';
+import type { ISubscribeArgs, ISubscribeResponse } from './types';
+
+export const subscribeToMasjid = async ({ email, topics }: ISubscribeArgs) => {
+	const url = new URL(PUBLIC_SUBSCRIPTIONS_BASE_URL);
+	url.searchParams.append('email', email);
+	for (const topic of topics) {
+		url.searchParams.append('topics', topic);
+	}
+
+	const response = await fetch(url, {
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/json',
+			Accept: 'application/json'
+		}
+	});
```

Comment:
@hammady I tried to do an AJAX POST request and encountered 2 issues:
1. The OPTION header sends an Accept '*/*' and receives a 302 redirect response. I know I asked to remove it but didn't think about this case 🫥 .. can we Remove the check for OPTIONS?
2. In order to work-around No.1 I tried sending `mode: no-cors` with the request, but I received 200 status without a response.


٢ز

Reply:
@fayez-nazzal The endpoint does not look at the http method and will always process the request in the same manner. I think we should exclude the `OPTIONS` and return a specific response without actually processing the request. Do we simply return 200 with empty body?

Reply:
@hammady Parsing the response as JSON or text fails, and when inspected in the network tab I get this message:

<img width="305" alt="image" src="https://github.com/hammady/wwpray/assets/49946791/f76d42f4-24f7-4d1f-9f3f-8073a0eed5e5">

<br />
<br />

<img width="341" alt="image" src="https://github.com/hammady/wwpray/assets/49946791/ab15d8a8-28df-47f5-baba-6b5efd62de87">

<br />
<br />

<img width="340" alt="image" src="https://github.com/hammady/wwpray/assets/49946791/d0a68ec4-7945-4136-a444-5b91eeba6c46">

Reply:
That's weird as I can see content-type and length set correctly in the response. Should I fix the OPTIONS then we try again?

Reply:
@hammady The last request didn't involve OPTIONS, maybe because of no-cors mode

Reply:
@hammady Could it be because of `Accept` and `Content-Type` in the request headers?

Reply:
@fayez-nazzal You don't really need the `Content-Type` in request headers because the request should be sending all params in the query string with an empty body. The empty body does not need any content type as such.
Did you try doing the request using curl or any other non-browser client?
So we don't need the `OPTIONS` support anymore?

Reply:
```
$ aws sesv2 delete-contact --email-address REDACTED@gmail.com --contact-list-name wwpray
(.venv)  hossam:wwpray $ curl 'https://lgigm92nn6.execute-api.us-east-1.amazonaws.com/?email=REDACTED@gmail.com&topics=Shalimar' -H 'Accept: application/json' -v
*   Trying 34.231.201.77:443...
* Connected to lgigm92nn6.execute-api.us-east-1.amazonaws.com (34.231.201.77) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=*.execute-api.us-east-1.amazonaws.com
*  start date: Jul 25 00:00:00 2023 GMT
*  expire date: Aug 21 23:59:59 2024 GMT
*  subjectAltName: host "lgigm92nn6.execute-api.us-east-1.amazonaws.com" matched cert's "*.execute-api.us-east-1.amazonaws.com"
*  issuer: C=US; O=Amazon; CN=Amazon RSA 2048 M02
*  SSL certificate verify ok.
* using HTTP/2
* h2h3 [:method: GET]
* h2h3 [:path: /?email=REDACTED@gmail.com&topics=Shalimar]
* h2h3 [:scheme: https]
* h2h3 [:authority: lgigm92nn6.execute-api.us-east-1.amazonaws.com]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: application/json]
* Using Stream ID: 1 (easy handle 0x130013600)
> GET /?email=REDACTED@gmail.com&topics=Shalimar HTTP/2
> Host: lgigm92nn6.execute-api.us-east-1.amazonaws.com
> user-agent: curl/7.88.1
> accept: application/json
> 
< HTTP/2 200 
< date: Sun, 15 Oct 2023 20:44:08 GMT
< content-type: application/json
< content-length: 139
< apigw-requestid: M3FLyjleoAMEJ0g=
< 
* Connection #0 to host lgigm92nn6.execute-api.us-east-1.amazonaws.com left intact
{"message": "Successfully subscribed to new topics. A confirmation email has been sent to REDACTED@gmail.com.", "topics": ["Shalimar"]}(.venv)  hossam:wwpray $ 
```

Reply:
@hammady Tried without `Content-Type`, same issue. If it works well with curl, it's a CORS issue. In this case, we need to support OPTIONS. The `no-cors` mode may be the reason why the response was unparsable.

I tried the curl request and got a successful response.

```zsh
curl 'https://lgigm92nn6.execute-api.us-east-1.amazonaws.com/?email=me@fayez.io&topics=Shalimar' -H 'Accept: application/json'

{"message": "Already subscribed to all topics.", "topics": []}%                                                                                       
```

Reply:
@hammady It works now!

Reply:
Did anything change?

Reply:
@hammady It's a simple request, not sending OPTIONS anymore

Reply:
I deployed a change to the function configuration but it is unrelated to this.
Check the latest commits.
If that's ready now I will approve and merge. We are getting closer!

Reply:
Making it a simple request solved it. Yes, it's ready. If everything goes well, feel free to merge. This morning will look into #20